build.sbt

```diff
@@ -2,379 +2,402 @@ import java.io._
-  publish := {
-    if (sys.props("sbt.prohibit.publish") != null) sys.error("Undefined publishing strategy")
-
-    // generate the scalatex readme into `website`
-    val website = new File(target.value.getAbsolutePath + File.separator + "scalatex")
-    if (website.exists) website.delete
-    val _ = (run in Compile).toTask(" --validate").value
-    if (!website.exists) sys.error("failed to generate the scalatex website")
-
-    // import the scalatex readme into `repo`
-    val repo = new File(os.temp.mkdir.getAbsolutePath + File.separator + "scalameta.org")
-    os.shell.call(s"git clone https://github.com/scalameta/scalameta.github.com ${repo.getAbsolutePath}")
-    println(s"erasing everything in ${repo.getAbsolutePath}...")
-    repo.listFiles.filter(f => f.getName != ".git").foreach(os.shutil.rmtree)
-    println(s"importing website from ${website.getAbsolutePath} to ${repo.getAbsolutePath}...")
-    new PrintWriter(new File(repo.getAbsolutePath + File.separator + "CNAME")) { write("scalameta.org"); close }
-    website.listFiles.foreach(src => os.shutil.copytree(src, new File(repo.getAbsolutePath + File.separator + src.getName)))
-
-    // commit and push the changes if any
-    os.shell.call(s"git add -A", cwd = repo.getAbsolutePath)
-    val nothingToCommit = "nothing to commit, working directory clean"
-    try {
-      val currentUrl = s"https://github.com/scalameta/scalameta/tree/" + os.git.currentSha()
-      os.shell.call(s"git config user.email 'scalametabot@gmail.com'", cwd = repo.getAbsolutePath)
-      os.shell.call(s"git config user.name 'Scalameta Bot'", cwd = repo.getAbsolutePath)
-      os.shell.call(s"git commit -m $currentUrl", cwd = repo.getAbsolutePath)
-      val httpAuthentication = os.secret.obtain("github").map{ case (username, password) => s"$username:$password@" }.getOrElse("")
-      val authenticatedUrl = s"https://${httpAuthentication}github.com/scalameta/scalameta.github.com"
-      os.shell.call(s"git push $authenticatedUrl master", cwd = repo.getAbsolutePath)
-    } catch {
-      case ex: Exception if ex.getMessage.contains(nothingToCommit) => println(nothingToCommit)
+  .dependsOn(common, dialects, inputs, tokens)
+lazy val tokenizersJVM = tokenizers.jvm
+lazy val tokenizersJS = tokenizers.js
+
+lazy val tokens = crossProject
+  .in(file("scalameta/tokens"))
+  .settings(
+    publishableSettings,
+    description := "Scala.meta's tokens and token-based abstractions (inputs and positions)",
+    enableMacros)
+  .dependsOn(common, dialects, inputs)
+lazy val tokensJVM = tokens.jvm
+lazy val tokensJS = tokens.js
+
+lazy val transversers = crossProject
+  .in(file("scalameta/transversers"))
+  .settings(
+    publishableSettings,
+    description := "Scala.meta's traversal and transformation infrastructure for abstract syntax trees",
+    enableMacros)
+  .dependsOn(common, trees)
+lazy val traversersJVM = transversers.jvm
+lazy val traversersJS = transversers.js
```

Comment:
The `lazy val blahJVM = blah.jvm; lazy val blahJS = blah.js` is getting a little bit old by now :) I wonder whether there is a possibility to abstract this away.